### PR TITLE
Use node ip autodiscovery [forked]

### PR DESCRIPTION
**Pull request by BenTheElder**
https://github.com/kubernetes-sigs/kind/pull/1513 but with the base image cleaned up.

### CODE REVIEW

File: images/base/Dockerfile [hunk 1]
```diff
@@ -108,7 +108,9 @@ RUN echo "Ensuring scripts are executable ..." \
  && echo "Ensuring /etc/kubernetes/manifests" \
     && mkdir -p /etc/kubernetes/manifests \
  && echo "Adjusting systemd-tmpfiles timer" \
-    && sed -i /usr/lib/systemd/system/systemd-tmpfiles-clean.timer -e 's#OnBootSec=.*#OnBootSec=1min#'
+    && sed -i /usr/lib/systemd/system/systemd-tmpfiles-clean.timer -e 's#OnBootSec=.*#OnBootSec=1min#' \
+ && echo "Modifying /etc/nsswitch.conf to prefer hosts" \
+    && sed -i /etc/nsswitch.conf -re 's#^(hosts:\s*).*#\1dns files#'
```
Comment (BenTheElder) [inline images/base/Dockerfile:113]: preserve the indent and do this at build time instead of runtime

**Comment by aojea** [comment]
Could you check if reboot works?

File: images/base/Dockerfile (hunk 1 above)
Comment (aojea) [reply images/base/Dockerfile:113]: This seems better indeed

**Comment by BenTheElder** [comment]
I didn't yet, and I'm currently booted to windows on my desktop 😱 /s (no IPv6 clusters on mac...)

**Comment by aojea** [comment]
no problem, I'm going to do it now

**Comment by aojea** [comment]
Before restart
```
ubectl get nodes -o wide
NAME                 STATUS   ROLES    AGE     VERSION                                    INTERNAL-IP   EXTERNAL-IP   OS-IMAG
E       KERNEL-VERSION                CONTAINER-RUNTIME
kind-control-plane   Ready    master   3m54s   v1.19.0-alpha.2.172+ce65f0e7ddeb60-dirty   172.19.0.2    <none>        Ubuntu 
19.10   4.12.14-lp151.28.13-default   containerd://1.3.3-14-g449e9269
kind-worker          Ready    <none>   3m19s   v1.19.0-alpha.2.172+ce65f0e7ddeb60-dirty   172.19.0.3    <none>        Ubuntu 
19.10   4.12.14-lp151.28.13-default   containerd://1.3.3-14-g449e9269
kind-worker2         Ready    <none>   3m19s   v1.19.0-alpha.2.172+ce65f0e7ddeb60-dirty   172.19.0.4    <none>        Ubuntu 
19.10   4.12.14-lp151.28.13-default   containerd://1.3.3-14-g449e9269

```

After only control-plane goes up, the workers fail with the following error

```
sed: can't read /etc/kubernetes/manifests/*.yaml: No such file or directory
```

**Comment by BenTheElder** [comment]
I can fix that in the morning I think.

**Comment by aojea** [comment]
> I can fix that in the morning I think.

don't worry, I'll send a patch

**Comment by aojea** [comment]
/hold

This has a side effect, now we have two ip addresses

```
root@kind-control-plane:/# hostname --ip-address
fc00:db8:2::3 172.19.0.3
```

**Comment by aojea** [comment]
and it always return ipv6 first, I just checked in an IPv4 node
```
docker exec -it tewst-control-plane hostname --ip-address
fc00:db8:2::5 172.19.0.5
```

**Comment by aojea** [comment]
it does work with ipv6 :partying_face:

**Comment by BenTheElder** [comment]
Can we pass kubelet 0.0.0.0 for ipv4 discovery?

**Comment by aojea** [comment]
> Can we pass kubelet 0.0.0.0 for ipv4 discovery?

the option was implemented only for `::`

I think is better to pass the `empty` value, I think that the option has `omitempty`

**Comment by BenTheElder** [comment]
superseded https://github.com/kubernetes-sigs/kind/pull/1521

**Pull request closed without merging**